pkg/types/azure/disk.go

```diff
@@ -0,0 +1,54 @@
+package azure
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var (
+	RxDiskEncryptionSetID   = regexp.MustCompile(`(?i)^/subscriptions/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})/resourceGroups/([-a-z0-9_().]{0,89}[-a-z0-9_()])/providers/Microsoft\.Compute/diskEncryptionSets/([-a-z0-9_]{1,80})$`)
+	RxSubscriptionID        = regexp.MustCompile(`^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$`)
+	RxResourceGroup         = regexp.MustCompile(`^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
+	RxDiskEncryptionSetName = regexp.MustCompile(`^[-a-z0-9_]{1,80}$`)
+)
+
+// ToID creates an Azure resource ID for the disk encryption set.
+func (d *DiskEncryptionSet) ToID() string {
+	return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/diskEncryptionSets/%s",
+		d.SubscriptionID, d.ResourceGroup, d.Name)
+}
```

Comment:
nit: This function can return an invalid id when subscription is empty. (but it won't in the current code flow so I don't see it as a blocker, just mention it in case you see a room for improvement).